PagerDuty/barkdog
# SRE-2043 Metric type fix

### Context
Datadog for some reason, decides that a given query is no longer type `query` but is now expected to be type `metric`. In the past, teams have needed to go and manually adjust their monitors. This was no good for myriad reasons.

Barkdog is a mid-step between `pd-datadog-monitors` and the datadog api. Since it was a public github repo, we at PagerDuty forked this repo, and unfortunately the test in the rspec are a tangled pile of garbage. 

### What Changed
This PR uses the code that Mia wrote on `ignore-fix-tags-like-metric-type` which, due to this being a change that is handled within Datadog itself, is untestable. 

To get the rspec tests to pass, Mia and I commented out the tests and ran each test one-by-one and then all together with the appropriate environment variables. Between that work and creating a new branch cherry-picked with Mia's changes, one part of a test continues to fail. Which is commented out in this PR.

### Why Aren't You Making that Test Pass without Commenting Parts Out?
Great question. This repo is very tangled, the test that fails without the comments is not a test that touches the code commented out, and the only failure is a text/json type error. Given how long these changes took to make and the un-test-able-ness of the entire changes, this engineer decided it was not worth the time to untangle this. 

### How Can I Test This PR?
Clone the forked repo, use sandbox environment variables, run `bundle exec rake rspec` to see their integrations test working. The logic of changing query type cannot be tested to my knowledge.
